File: 0_mraa_intro/3_Isr.js
/*eslint no-console:0*/

var m = require('mraa');
console.log('MRAA Version: ' + m.getVersion()); //write the mraa version to the console

function encodeL() {
  c --;
  console.log(c);
}

function encodeR() {
  c ++;
  console.log(c);
}

var l = new m.Gpio(6);
var r = new m.Gpio(7);
var c = 0;

l.dir(m.DIR_IN);
l.mode(m.MODE_PULLUP);
r.dir(m.DIR_IN);
r.mode(m.MODE_PULLUP);

l.isr(m.EDGE_RISING, encodeL);
r.isr(m.EDGE_FALLING, encodeR);

setInterval(function(){
  /* nop */
},1000);

File: 0_mraa_intro/3_Isr.py
import mraa
import time


class Counter:
    count = 0


def encodeL(args):
    c.count -= 1
    print(c.count)


def encodeR(args):
    c.count += 1
    print(c.count)


print('MRAA Version: %s' % mraa.getVersion())

l = mraa.Gpio(6)
r = mraa.Gpio(7)
c = Counter()

l.dir(mraa.DIR_IN)
l.mode(mraa.MODE_PULLUP)
r.dir(mraa.DIR_IN)
r.mode(mraa.MODE_PULLUP)

l.isr(mraa.EDGE_RISING, encodeL, encodeL)
r.isr(mraa.EDGE_FALLING, encodeR, encodeR)

time.sleep(1000)
